lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php

```diff
@@ -25,6 +25,64 @@ public function setForceRequest($forced)
         return $this->forceRequest = $forced;
     }
 
+    public function parameters()
+    {
+        $params = parent::parameters();
+
+
+        // Filter data by login if necessary
+        // as configured in the plugin for login filtered layers.
+
+        // Filter data by login for request: getmap, getfeatureinfo, getprint, getprintatlas
+        if (!in_array($this->param('request'), array('getmap', 'getfeatureinfo', 'getprint', 'getprintatlas'))) {
+            return $params;
+        }
+
+        // No filter data by login rights
+        if (jAcl2::check('lizmap.tools.loginFilteredLayers.override', $this->repository->getKey())) {
+            return $params;
+        }
+
+        // filter data by login
+        $layers = $this->param('layers');
+        if (is_string($layers)) {
+            $layers = explode(',', $layers);
+        }
+
+        // get login filters
+        $loginFilters = $this->project->getLoginFilters($layers);
+
+        // login filters array is empty
+        if (empty($loginFilters)) {
+            return $params;
+        }
+
+        // merge client filter parameter
+        $clientFilter = $this->param('filter');
+        if ($clientFilter != Null && !empty($clientFilter)) {
+            $cfexp = explode(';', $clientFilter);
+            foreach ($cfexp as $a) {
+                $b = explode(':', $a);
+                $lname = trim($b[0]);
+                $lfilter = trim($b[1]);
+                if (array_key_exists($lname, $loginFilters)) {
+                    $loginFilters[$lname]['filter'] .= ' AND '.$lfilter;
+                } else {
+                    $loginFilters[$lname] = array('filter' => $lfilter, 'layername' => $lname);
+                }
+            }
+        }
+
+        // update filter parameter
+        $filters = array();
+        foreach ($loginFilters as $layername => $lfilter) {
+            $filters[] = $layername.':'.$lfilter['filter'];
```

Comment:
notice	2020-07-24 15:16:07	[8]	Trying to access array offset on value of type null	/lizmap_web_client/lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php	79